### DISCUSSION TITLE
Update to use on UE 5.3

### DISCUSSION THREAD

**Discussion by Benjf29** [hfjooste/UltimateStarterKit#108] [Ideas]
Hi,

I'm very interested about using your starter kit. I'm currently using UE 5.3 and the kit are compiled for 5.2 max. It is possible for me to, from source, recompile for the 5.3 ? Or, the best option, if you can build it for 5.3

Thank, and really nice work ! :-)

**Comment by hfjooste** [comment]
Hey @Benjf29. Thanks. Appreciate the feedback!

There is an open [issue](https://github.com/hfjooste/UltimateStarterKit/issues/62) to add support for UE5.3. I looked at this when 5.3 was first released. There is some code I need to change, and I have no idea how to get the input remapping to work. The docs were really bad when I tried to add UE 5.3 support.

I'll create a `ue5.3` branch and add the changes there. You should be able to use everything on that branch except input remapping. As soon as I have time to figure that out, I'll create an official release with UE 5.3 support

**Reply by hfjooste** [reply]
Actually, that will make it more difficult for to use since you'll have to compile everything yourself. I'll just create a new release that supports UE5.3 and open an issue to fix the remapping. I'm working on the release now and should be available in the next few hours

**Reply by Benjf29** [reply]
Hi,
I didn't see the existing issue, sorry about that.
I'm really happy to hear that. I've see the 3.13  version and try it, but I get the same error : "The following modules are missing or built with a different engine version: USK USKEditor". It is not because the EngineAssociation are set to 5.2 in uproject file ?

Thank for you'r time :-)

**Reply by hfjooste** [reply]
@Benjf29 Did you mean 3.12? (There isn't a 3.13). The demo project is still using 5.2. I'm actually planning to downgrade that to 4.27 so that I can make sure all the assets work properly for a marketplace release. That shouldn't affect the plugin.

The plugin should work on 5.3 if you download this one: [v3.12.0 (UE 5.3)](https://github.com/hfjooste/UltimateStarterKit/releases/download/3.12.0/UltimateStarterKit-UE5.3-v3.12.0.zip)

Here's a screenshot in one of my UE5.3 projects using the plugin
![image](https://github.com/hfjooste/UltimateStarterKit/assets/18502706/b0f0e7dc-172a-40bd-905f-163e20602302)

**Reply by Benjf29** [reply]
Yeah, 3.12, sorry I'm tired
I still getting the same message with this version 
I'm on Mac, that's the difference, not good news 
<img width="725" alt="Capture d’écran 2023-10-29 à 19 08 46" src="https://github.com/hfjooste/UltimateStarterKit/assets/22636985/01fc79a4-b562-4f55-85f6-48b68e42b8a0">

**Reply by hfjooste** [reply]
Yeah I've never tried it on Mac. I don't have one and can't really debug or fix those issues. Can you try to copy the entire folder to the Plugins folder in the root of your project? (Create one if it doesn't exist). Then generate the project file and try to build it through your IDE? Maybe that will work or at least give us a better error that I can try to fix

**Reply by Benjf29** [reply]
<img width="258" alt="Capture d’écran 2023-10-29 à 19 24 51" src="https://github.com/hfjooste/UltimateStarterKit/assets/22636985/3dacc47c-959f-46cc-a2b8-e79592b85c93">
Got this with the folder in project/Plugins and try to rebuild it, a not really explicit message

**Reply by hfjooste** [reply]
But try to start it through the IDE (Visual Studio, Rider or whatever you're using on Mac). That should tell you exactly what is wrong in the source code. Sorry I really don't know what else to try without doing it myself on a Mac
